crates/api_models/src/payments.rs

```diff
@@ -212,6 +212,9 @@ pub struct PaymentsRequest {
     /// Merchant connector details used to make payments.
     pub merchant_connector_details: Option<admin::MerchantConnectorDetailsWrap>,
 
+    /// Allowed Payment Method Types for a given PaymentIntent
+    pub allowed_payment_method_types: Option<Vec<api_enums::PaymentMethodType>>,
```

Comment:
```suggestion
    #[schema(value_type = Option<Vec<PaymentMethodType>>)]
    pub allowed_payment_method_types: Option<Vec<api_enums::PaymentMethodType>>,
```

This so that the OpenAPI spec generated is proper.